docs/product/guidelines/theming-new.html

```diff
@@ -0,0 +1,537 @@
+    .create-custom-theme-hsl-rgb-variables(hsl(172, 37%, 48%), primary, dark);
+    .create-custom-theme-hsl-rgb-variables(hsl(259, 29%, 55%), secondary, dark);
+}
+{% endhighlight %}
+    </div>
+
+        {% header "h4", "Output" %}
+
+    <p class="stacks-copy">Explain how the above function the variables below and how a consumer can opt to forgo the function and just set the variables. Make sure to describe the base/dark and primary/secondary variants.</p>
+
+    <div class="stacks-preview _preview-none mb24">
+{% highlight css %}
+/* Input */
+.theme-custom.themed {
+    .create-custom-theme-hsl-rgb-variables(hsl(172, 37%, 48%), primary);
+}
+
+/* Output */
+.theme-custom.themed {
+    /* HSL variables */
+    --theme-base-secondary-color-new-h: 172;
+    --theme-base-secondary-color-new-s: 37%;
+    --theme-base-secondary-color-new-l: 48%;
+    /* RGB variables */
+    --theme-base-secondary-color-new-r: 77.112;
+    --theme-base-secondary-color-new-g: 167.688;
+    --theme-base-secondary-color-new-b: 155.6112;
+    /* Color variables based on HSL and RGB variables */
+    --theme-primary-new-custom: hsl(var(--theme-base-primary-color-new-h), var(--theme-base-primary-color-new-s), var(--theme-base-primary-color-new-l));
+    --theme-primary-new-custom-100: rgb(calc(198.9 + var(--theme-base-primary-color-new-r) * 0.22), calc(198.9 + var(--theme-base-primary-color-new-g) * 0.22), calc(198.9 + var(--theme-base-primary-color-new-b) * 0.22));
+    --theme-primary-new-custom-200: rgb(calc(122.4 + var(--theme-base-primary-color-new-r) * 0.52), calc(122.4 + var(--theme-base-primary-color-new-g) * 0.52), calc(122.4 + var(--theme-base-primary-color-new-b) * 0.52));
+    --theme-primary-new-custom-300: rgb(calc(89.25 + var(--theme-base-primary-color-new-r) * 0.65), calc(89.25 + var(--theme-base-primary-color-new-g) * 0.65), calc(89.25 + var(--theme-base-primary-color-new-b) * 0.65));
+    --theme-primary-new-custom-400: var(--theme-primary-new-custom);
+    --theme-primary-new-custom-500: hsl(var(--theme-base-primary-color-new-h), var(--theme-base-primary-color-new-s), calc(var(--theme-base-primary-color-new-l) - 5%));
+    --theme-primary-new-custom-600: hsl(var(--theme-base-primary-color-new-h), var(--theme-base-primary-color-new-s), calc(var(--theme-base-primary-color-new-l) - 10%));
+}
+{% endhighlight %}
+    </div>
+
+    {% header "h4", "Manual addition of theme variables" %}
+    <p class="stacks-copy">If you need to apply a theme without using the above function, you can do so by manually adding the variables above to your CSS. The most common use for this approach is when the theme needs to change client-side, such as when allowing the user to change and preview a theme dynamically.</p>
+
+    <p class="stacks-copy">With this approach, we recommend targeting new h/s/l and r/g/b color variables on a parent element that includes <code class="stacks-code ws-nowrap">.themed</code> class.</p>
```

Comment:
It _honestly_ does not make a lot of sense to me either. I'd be open to changing this but generally, I yielded on the site of keeping these sorts of things the same to limit risk (and partially out of laziness since I didn't want to have to rewrite our `.native-tint` and `.native-shade` Less functions).

Do you think it would be worth it to change now?

Reply:
Similarly to the comment I made above we should at least have a place where we track all these tradeoffs we make to reduce liability and amount of work. 
I think it's ok to leave things as they are for this PR but it will be important to have a conversation about those "questionable" APIs **before** releasing 2.0. 
Can you add a point for this in [this document](https://docs.google.com/document/d/18JkYEVioPhPmHGZaV-ba8S-3rl_zdbMIhpqtsX7iqYk/edit?usp=sharing)? Thanks 🙂